utils/CMakeLists.txt

```diff
@@ -19,18 +19,21 @@ endforeach()
 
 # the debufr utility uses the "8_DA" (8-byte integer, 8-byte real, dynamic allocation)
 # build of the library
-#set(_lib debufr.o)
-#add_library(${_lib} OBJECT debufr.f)
-#set_target_properties(${_lib} PROPERTIES COMPILE_FLAGS
-#                                         "${fortran_8_DA_flags}")
-#set(_exec debufr.x)
-#add_executable(${_exec} debufr.c)
-#target_compile_definitions(${_exec} PUBLIC "${underscore_def}")
-#target_compile_definitions(${_exec} PUBLIC "${c_8_DA_defs}")
-#add_dependencies(${_exec} bufr::bufr_8_DA)
-#target_link_libraries(${_exec} PRIVATE $<TARGET_OBJECTS:${_lib}>)
-#target_link_libraries(${_exec} PRIVATE bufr::bufr_8_DA)
-#list(APPEND _utils_execs ${_exec})
+set(_exec debufr.x)
+add_executable(${_exec} debufr.c debufr.f)
+add_dependencies(${_exec} bufr::bufr_8_DA)
+set_property(SOURCE debufr.c APPEND_STRING PROPERTY COMPILE_DEFINITIONS
+                                                    "${underscore_def}"
+                                                    "${c_8_DA_defs}")
+set_property(SOURCE debufr.f APPEND_STRING PROPERTY COMPILE_FLAGS
+                                                    "${fortran_8_DA_flags}")
+target_include_directories(${_exec} PRIVATE "${PROJECT_BINARY_DIR}/src")
+set_property(TARGET ${_exec} PROPERTY LINKER_LANGUAGE C)
+target_link_libraries(${_exec} PRIVATE bufr::bufr_8_DA)
+if(CMAKE_C_COMPILER_ID MATCHES "^(Intel)$")
+  target_link_options(${_exec} PRIVATE "-nofor-main")
+endif()
```

Comment:
Can you post the message you are getting when you remove this if-block?
I am building with Intel and see no errors or warnings and tests pass.

Reply:
The message doesn't show up on Intel; rather, it shows up on any other compiler other than Intel.  For example:

[100%] Linking C executable debufr.x
gcc: error: unrecognized command-line option '-nofor-main'; did you mean '-Wno-main'?
make[2]: *** [utils/CMakeFiles/debufr.x.dir/build.make:119: utils/debufr.x] Error 1
make[1]: *** [CMakeFiles/Makefile2:899: utils/CMakeFiles/debufr.x.dir/all] Error 2

Reply:
Otherwise, without the "-nofor-main" flag on Intel, the error message I was seeing earlier is already shown above, in a previous reply on this thread.

*However*, and for whatever weird reason, when I just tried to rebuild it again, it now works fine under Intel on the WCOSS.  Only thing I can figure is that some of the other edits we recently made must have inadvertently fixed whatever the problem is that was occurring earlier.

So I'll go ahead and push up yet another commit...